engine/apps/api/serializers/paging.py

```diff
@@ -0,0 +1,68 @@
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import serializers
+
+from apps.alerts.models import AlertGroup
+
+
+class UserReferenceSerializer(serializers.Serializer):
+    id = serializers.CharField()
+    important = serializers.BooleanField()
+    instance = serializers.HiddenField(default=None)  # set in UserReferenceSerializer.validate
+
+    def validate(self, attrs):
+        organization = self.context["organization"]
+
+        try:
+            attrs["instance"] = organization.users.get(public_primary_key=attrs["id"])
+        except ObjectDoesNotExist:
+            raise serializers.ValidationError("User {} does not exist".format(attrs["id"]))
+
+        return attrs
+
+
+class ScheduleReferenceSerializer(serializers.Serializer):
+    id = serializers.CharField()
+    important = serializers.BooleanField()
+    instance = serializers.HiddenField(default=None)  # set in ScheduleReferenceSerializer.validate
+
+    def validate(self, attrs):
+        organization = self.context["organization"]
+
+        try:
+            attrs["instance"] = organization.oncall_schedules.get(public_primary_key=attrs["id"])
+        except ObjectDoesNotExist:
+            raise serializers.ValidationError("Schedule {} does not exist".format(attrs["id"]))
+
+        return attrs
+
+
+class DirectPagingSerializer(serializers.Serializer):
+    users = UserReferenceSerializer(many=True, required=False, default=list)
+    schedules = ScheduleReferenceSerializer(many=True, required=False, default=list)
+
+    alert_group_id = serializers.CharField(required=False, default=None)
+    alert_group = serializers.HiddenField(default=None)  # set in DirectPagingSerializer.validate
+
+    title = serializers.CharField(required=False, default=None)
+    message = serializers.CharField(required=False, default=None)
+
+    def validate(self, attrs):
+        if len(attrs["users"]) == 0 and len(attrs["schedules"]) == 0:
+            raise serializers.ValidationError("Provide at least one user or schedule")
+
+        if not attrs["alert_group_id"] and not attrs["title"]:
+            raise serializers.ValidationError("Provide either alert_group_id or title")
```

Comment:
I think the idea was to fallback to a default title if none is provided? (the backend logic will set one if this is `None`).

Reply:
https://github.com/grafana/oncall/pull/1098/commits/55d248f9e612e1c47411f2ea2da27520d6acdee7